engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -58,7 +60,10 @@ class PeriodicProcessService(delegateProcessManager: ProcessManager,
 
     def handleSingleProcess(deployedProcess: PeriodicProcessDeployment): Future[Unit] = {
       delegateProcessManager.findJobStatus(deployedProcess.periodicProcess.processVersion.processName).flatMap { state =>
-        handleFinished(deployedProcess, state).runWithCallbacks
+        handleFinishedAction(deployedProcess, state)
+          .flatMap { finishedSuccessfully =>
+            if (finishedSuccessfully) reschedule(deployedProcess, state) else scheduledProcessesRepository.monad.pure(()).emptyCallback
```

Comment:
maybe let's call it "needsReschedule"? 
```
 case _ => scheduledProcessesRepository.monad.pure(()).finished(successfully = false)
```  => this is not really unsuccessful, just unfinished? :)

Reply:
yes, it means unfinished in this case.

I've change name of this type alias to NeedsReschedule